kubert/src/runtime.rs

```diff
@@ -291,10 +321,14 @@ impl<S> Runtime<S> {
         watcher_config: watcher::Config,
     ) -> impl Stream<Item = watcher::Event<T>>
     where
-        T: Resource + DeserializeOwned + Clone + Debug + Send + 'static,
-        T::DynamicType: Default,
+        T: Resource<DynamicType = ()> + DeserializeOwned + Clone + Debug + Send + 'static,
```

Comment:
I don't think we need to make this backwards-incompatible change in type constraints. Instead, we can relax the type constraints used in the instrument_watch stream.